### PR TITLE
 #51 add memory

### PR DESCRIPTION
**Pull request by zhych1005**
#51

### CODE REVIEW

File: spring-ai-alibaba-core/pom.xml [hunk 1]
```diff
@@ -105,6 +105,50 @@
...
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-jdbc</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-redis</artifactId>
+		</dependency>
+
+		<dependency>
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/pom.xml:119]: 确定使用这个 orm 框架，或许可以考虑下 mybatis-flux？

File: spring-ai-alibaba-core/pom.xml [hunk 2]
```diff
@@ -105,6 +105,50 @@
...
+			<artifactId>mysql-connector-j</artifactId>
+			<version>8.0.33</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-elasticsearch</artifactId>
+		</dependency>
+
+		<dependency>
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/pom.xml:136]: 尽量别用hutool吧

File: spring-ai-alibaba-core/pom.xml [hunk 3]
```diff
@@ -105,6 +105,50 @@
...
+			<groupId>cn.hutool</groupId>
+			<artifactId>hutool-core</artifactId>
+			<version>5.7.5</version>
+		</dependency>
+
+		<dependency>
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/pom.xml:142]: 可以用 spring boot 自己的序列化框架 jackson

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/memory/entity/ChatMemoryProperties.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.alibaba.cloud.ai.memory.entity;
+
+import lombok.Data;
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/memory/entity/ChatMemoryProperties.java:3]: lombok 也别在基架项目里用

**Comment by chickenlj** [comment]
Well done.

I had a quick review of the code and overall it is what we expect to add for the chat memory feature. The main problem I want to point out is the boundaries of what the framework should do and what the user application should do. As a framework, we need to focus more on basic memory-related component definitions and make sure they are extensible.

**Pull request closed without merging**